guide/14-deep-learning/how_estimate_batch_size_works.ipynb

```diff
@@ -0,0 +1,251 @@
+{
```

Comment:
<p>This guide explains the steps for calculating the&nbsp;<strong>recommended_batchsize</strong>&nbsp;and&nbsp;<strong>max_batchsize</strong>&nbsp;of multiple network architectures supported by the&nbsp;<code>arcgis.learn</code>&nbsp;API. The&nbsp;<code>estimate_batch_size</code>&nbsp;function is designed to determine the optimal batch size for a deep learning model. Batch size is a critical hyperparameter in training deep learning models, and finding the right value can significantly impact the model's performance. The function takes in an&nbsp;<code>arcgis.learn</code>&nbsp;imagery model instance for which a batch size should be estimated and returns a batch size based on GPU capacity, size of the model, and the data used. By using the&nbsp;<code>estimate_batch_size</code>&nbsp;function, deep learning practitioners can prevent memory errors that can occur during training when the batch size is set too high. By finding the optimal batch size for the given model and system resources, the function can help ensure that the model fits within the available memory capacity and prevent errors that can interrupt the training process. This can save time and resources by avoiding the need to manually adjust the batch size or troubleshoot memory-related issues.</p><p>Currently only <code>arcgis.learn</code> imagery models are supported by the function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1533/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Reply:
<p>Added</p>